internal/testing/e2e/gateway.go

```diff
@@ -40,7 +40,7 @@ func (p *gatewayTestEnvironment) run(ctx context.Context, namespace string, cfg
 	consulClient := ConsulClient(ctx)
 
 	// this should go away once we implement auth in the server bootup
-	consulClient.AddHeader("x-consul-token", ConsulInitialManagementToken(ctx))
+	consulClient.Internal().AddHeader("x-consul-token", ConsulInitialManagementToken(ctx))
```

Comment:
Is this still needed at all now that we're using the consul-server-connection-manager integrated login flow?

Reply:
Yes, this is the client that gets passed around with the root auth token in our tests so that we can actually check what's in Consul.